src/ja/worker/config.py

```diff
@@ -10,27 +9,73 @@ class WorkerConfig(Config):
     Config for the worker client.
     """
 
+    def __init__(self, uid: str, ssh_config: SSHConfig, resource_allocation: ResourceAllocation):
+        """!
+        creates resource allocation instance
+        """
+        if uid is None:
+            raise ValueError("missing uid")
+        if ssh_config is None:
+            raise ValueError("missing ssh config")
+        if resource_allocation is None:
+            raise ValueError("missing resources")
+
+        self._ssh_config = ssh_config
+        self._resource_allocation = resource_allocation
+        self._uid = uid
+
+    def __eq__(self, o: object) -> bool:
+        if isinstance(o, WorkerConfig):
+            return self._ssh_config == o._ssh_config \
+                and self._uid == o._uid \
+                and self._resource_allocation == o._resource_allocation
+        else:
+            return False
+
     @property
     def uid(self) -> str:
```

Comment:
@JohannesGaessler why is there a UID in the worker? Isn't this assigned by the server? Or is this a label? I can imagine this being useful if you could specify a label like "room100_" and then UIDs are generated in the server as "room100_1, room100_2", etc.

Reply:
The worker needs to be able to set its own UID to be able to have a persistent UID upon program restart/system reboot.
The idea is that the worker will remember its UID so that it will register with the same UID the next time you connect to the server.
Otherwise the same worker registering to the server multiple times would have a different UID each time.
Or even worse, the workers would swap UIDs between each other.